specifyweb/frontend/js_src/lib/components/QueryBuilder/Wrapped.tsx

```diff
@@ -149,8 +155,17 @@ export function QueryBuilder({
     'showHiddenFields'
   );
 
-  const isResourceModified = useIsModified(queryResource);
-  const saveRequired = isResourceModified || state.saveRequired;
+  const [saveRequired, setSaveRequired] = React.useState(false);
+  const checkForChanges = throttle(() => {
+    const initialFieldsObject = JSON.parse(initialFields.current);
```

Comment:
because that will always return true otherwise

Reply:
then you have a bug somewhere as that's not supposed to be the case
if you are comparing different objects, then yeah, it would always be different.
but if you converted each to string before (using JSON.stringify() then it should compare correctly